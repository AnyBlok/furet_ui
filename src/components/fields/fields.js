export const fields = {
  list: {},
  form: {},
  thumbnail: {},
  kanban: {},
};
export default {
    fields,
}
