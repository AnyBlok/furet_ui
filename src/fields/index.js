/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import React from 'react';
import plugin from '../plugin';
import './string';
import './selection';
import './date';
import './integer';
import './float';
import './url';
import './uuid';
import './sequence';
import './password';
import './color';
import './text';

/**
 * Unknown field, used only if the wanted field is not available
**/
plugin.set(['field'], {Unknown: (props) => {
    return <div>{props.value}</div>
}});

/**
 * return the component for viewType and fieldType
**/
export const getField = (viewType, fieldType, attribs, value) => {
    let field = plugin.get(['field', viewType, fieldType]);
    if (!field) {
        field = plugin.get(['field', 'Unknown']);
        console.log('getField', viewType, fieldType, attribs, value)
    }
    const id = 'view-type-' + viewType + '-field_type-' + fieldType + '-field-name-' + (attribs.name || '');
    return React.createElement(field, Object.assign({id}, attribs, {value}));
}

export default {
    getField,
}
