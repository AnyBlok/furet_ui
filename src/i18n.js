/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueI18n from 'vue-i18n';
Vue.use(VueI18n);


export const i18nConf = {
    locale: 'en',
    fallbackLocale: 'en',
    messages: {
        en: {
            menus: {
                close: 'Close',
                search: 'Search ...',
            },
            views: {
                unknown: {
                    title: 'The wanted view "{name}" is unknown',
                    message: "Report this message to the administrator",
                },
                common: {
                    create: 'Create',
                    save: 'Save',
                    edit: 'Edit',
                    cancel: 'Cancel',
                    delete: 'Delete',
                    close: 'Close',
                    actions: 'Actions',
                    more: 'More',
                },
                clients: {
                    login: {
                        button: 'Log in',
                    },
                },
            },
            fields: {
                common: {
                    required: 'this field is required',
                },
                json: {
                    invalid: 'Invalid JSON format',
                },
                file: {
                    upload: 'Upload',
                    download: 'Download',
                    delete: 'Delete',
                },
            },
        },
    },
};
export const i18n = new VueI18n(i18nConf);
