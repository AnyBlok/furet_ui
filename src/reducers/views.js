/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
export const defaultState = {}

export const views = (state = defaultState, view) => {
    const value = Object.assign({}, view);
    delete value.type;
    switch (view.type) {
        case 'UPDATE_VIEW':
            delete value.viewId;
            const values = {};
            values[view.viewId] = Object.assign({}, state[view.viewId], value);
            return Object.assign({}, state, values);
        case 'CLEAR_ACTION':
            return defaultState;
        default:
            return state
    }
}

export default views;
