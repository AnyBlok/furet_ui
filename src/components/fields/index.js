import { fields } from './fields';
import './common';
import './boolean';
import './datetime';
import './integer';
import './password';
import './timestamp';
import './selection';
import './string';

import './relationship';

import { defineComponent } from '../factory';

defineComponent('furet-ui-unknown-field', {
  template : `<div>Unknow Field <strong>{{config.type}}</strong></div>`,
  prototype: {
    props: ['config', 'data'],
  }
})

defineComponent('furet-ui-field', {
  template : `
    <component 
      v-bind:is="component" 
      v-bind:resource="resource" 
      v-bind:config="config" 
      v-bind:data="data"
    />
  `,
  prototype: {
    props: ['resource', 'config', 'data'],
    computed: {
      component () {
        return (fields[this.resource.type] || {})[this.config.type] || 'furet-ui-unknown-field'
      },
    },
  }
})
