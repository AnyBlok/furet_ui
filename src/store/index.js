/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import _ from 'underscore';
import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import modules from './modules';
import {i18n} from '../i18n';
import {router} from '../routes';
import moment from 'moment';

Vue.use(Vuex)
const debug = process.env.NODE_ENV !== 'production'

export const storeDef = {
    actions,
    getters,
    modules,
    strict: debug,
};

export const store = new Vuex.Store(storeDef);

export const dispatchAll = (datas) => {
    _.each(datas, data => {
        if (data.type) {
            switch (data.type) {
                case 'UPDATE_LOCALES':
                    _.each(data.locales, locale => {
                        i18n.setLocaleMessage(locale.locale, locale.messages)
                    });
                    break
                case 'SET_LOCALE':
                    i18n.locale = data.locale;
                    moment.locale(data.locale);
                    break
                case 'UPDATE_ROUTE':
                    if (data.name) router.push({name: data.name, params: data.params});
                    else if (data.path) router.push({path: data.path});
                    break;
                default:
                    const type = data.type;
                    delete data.type;
                    const process = data.process || 'commit';
                    store[process](type, data);
            }
        }
    });
}

export default store;
