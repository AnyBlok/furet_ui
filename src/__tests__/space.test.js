/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import Buefy from 'buefy';
Vue.use(Buefy, {defaultIconPack: 'fa',});
import {store} from '../store';
import {router} from '../routes';
import {i18n} from '../i18n';
import '../view';
import '../views';
import '../fields';
import {changeView, onClickBreadScrumb, onClickMenu, ViewSelector, SpaceMenu, Space} from '../space';

jest.mock('../server-call')
const menus = [
    {
        label: 'Customer',
        image: {'type': 'font-icon', 'value': 'fa-user'},
        actionId: '2',
        id: '1',
        submenus: [],
    },
    {
        label: 'Setting',
        image: {'type': 'font-icon', 'value': 'fa-user'},
        actionId: '',
        id: '2',
        submenus: [
            {
                label: 'Category',
                image: {'type': '', 'value': ''},
                actionId: '3',
                id: '3',
                submenus: [],
            },
            {
                label: 'Address',
                image: {'type': '', 'value': ''},
                actionId: '4',
                id: '4',
                submenus: [],
            },
        ],
    },
]

describe('View Selector component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
        router.push({path: '/'});
    });
    it('Render', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space, {params: {
                viewId: '1',
                views: [
                    {viewId: '1', type: 'List'},
                    {viewId: '2', type: 'Form', unclickable: true},
                ],
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render unclickable selected', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space, {params: {
                viewId: '2',
                views: [
                    {viewId: '1', type: 'List'},
                    {viewId: '2', type: 'Form', unclickable: true},
                ],
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('SpaceMenu component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
        router.push({path: '/'});
    });
    it('Render', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(SpaceMenu),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with submenu', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(SpaceMenu, {props: {
                menuId: '1',
                menus,
                spaceId: '1',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with submenu other selected menu', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(SpaceMenu, {props: {
                menuId: '3',
                menus,
                spaceId: '1',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render change menu (with action)', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(SpaceMenu)
        });
        store.commit({type: 'REPLACE_CHANGE', changes: {test: 1}});
        store.commit({
            type: 'ADD_IN_BREADSCRUMB',
            path: 'space/1/action/3',
            label: 'Action 3',
            changes: {action: 3}
        });
        store.commit({
            type: 'ADD_IN_BREADSCRUMB',
            path: 'space/1/action/4',
            label: 'Action 4',
            changes: {action: 4}
        });
        router.push({path: '/'});
        expect(store.state.data.changes.test).toBe(1);
        expect(store.state.global.breadscrumbs.length).toBe(2);
        expect(vm.$route.path).toBe('/');
        onClickMenu(router, '1', {id: '2', actionId: '3'})
        expect(store.state.data.changes.test).toBe(undefined);
        expect(vm.$route.path).toBe('/space/1/menu/2/action/3');
        expect(store.state.global.breadscrumbs.length).toBe(0);
    });
    it('Render change menu (without action)', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(SpaceMenu)
        });
        store.commit({type: 'REPLACE_CHANGE', changes: {test: 1}});
        store.commit({
            type: 'ADD_IN_BREADSCRUMB',
            path: 'space/1/action/3',
            label: 'Action 3',
            changes: {action: 3}
        });
        store.commit({
            type: 'ADD_IN_BREADSCRUMB',
            path: 'space/1/action/4',
            label: 'Action 4',
            changes: {action: 4}
        });
        router.push({path: '/'});
        expect(store.state.data.changes.test).toBe(1);
        expect(store.state.global.breadscrumbs.length).toBe(2);
        expect(vm.$route.path).toBe('/');
        onClickMenu(router, '1', {id: '2'})
        expect(store.state.data.changes.test).toBe(undefined);
        expect(vm.$route.path).toBe('/');
        expect(store.state.global.breadscrumbs.length).toBe(0);
    });
});

describe('Space component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
        router.push({path: '/'});
    });
    it('Render', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render display left menu (close)', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space, {props: {
                spaceId: '1',
                menuId: '1',
                actionId: '1',
                viewId: '1',
            }}),
        });
        store.commit({
            type: 'UPDATE_SPACE',
            spaceId: '1',
            left_menu: menus,
            right_menu: [],
        });
        store.commit({                                                                       
            type: 'UPDATE_ACTION',
            actionId: '1',
            label: 'Action : 1',
            views: [
                {
                    viewId: '1',
                    type: 'List',
                },
                {
                    viewId: '2',
                    type: 'Thumbnail',
                },
                {
                    viewId: '3',
                    type: 'Form',
                    unclickable: true,
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render display left menu (open)', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space, {props: {
                spaceId: '1',
                menuId: '1',
                actionId: '1',
                viewId: '1',
                defaultOpenLeft: true,
            }}),
        });
        store.commit({
            type: 'UPDATE_SPACE',
            spaceId: '1',
            left_menu: menus,
            right_menu: [],
        });
        store.commit({                                                                       
            type: 'UPDATE_ACTION',
            actionId: '1',
            label: 'Action : 1',
            views: [
                {
                    viewId: '1',
                    type: 'List',
                },
                {
                    viewId: '2',
                    type: 'Thumbnail',
                },
                {
                    viewId: '3',
                    type: 'Form',
                    unclickable: true,
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render display right menu (close)', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space, {props: {
                spaceId: '1',
                menuId: '1',
                actionId: '1',
                viewId: '1',
            }}),
        });
        store.commit({
            type: 'UPDATE_SPACE',
            spaceId: '1',
            left_menu: [],
            right_menu: menus,
        });
        store.commit({                                                                       
            type: 'UPDATE_ACTION',
            actionId: '1',
            label: 'Action : 1',
            views: [
                {
                    viewId: '1',
                    type: 'List',
                },
                {
                    viewId: '2',
                    type: 'Thumbnail',
                },
                {
                    viewId: '3',
                    type: 'Form',
                    unclickable: true,
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render display right menu (open)', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space, {props: {
                spaceId: '1',
                menuId: '1',
                actionId: '1',
                viewId: '1',
                defaultOpenRight: true,
            }}),
        });
        store.commit({
            type: 'UPDATE_SPACE',
            spaceId: '1',
            left_menu: [],
            right_menu: menus,
        });
        store.commit({                                                                       
            type: 'UPDATE_ACTION',
            actionId: '1',
            label: 'Action : 1',
            views: [
                {
                    viewId: '1',
                    type: 'List',
                },
                {
                    viewId: '2',
                    type: 'Thumbnail',
                },
                {
                    viewId: '3',
                    type: 'Form',
                    unclickable: true,
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render breadscrumb', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space, {props: {
                spaceId: '1',
                menuId: '1',
                actionId: '1',
                viewId: '1',
            }}),
        });
        store.commit({
            type: 'UPDATE_SPACE',
            spaceId: '1',
            left_menu: [],
            right_menu: [],
        });
        store.commit({                                                                       
            type: 'UPDATE_ACTION',
            actionId: '1',
            label: 'Action : 1',
            views: [
                {
                    viewId: '1',
                    type: 'List',
                },
            ],
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        store.commit({
            type: 'ADD_IN_BREADSCRUMB',
            path: 'space/1/action/2',
            label: 'Action 2',
            changes: {action: 2}
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        store.commit({
            type: 'ADD_IN_BREADSCRUMB',
            path: 'space/1/action/3',
            label: 'Action 3',
            changes: {action: 3}
        });
        store.commit({
            type: 'ADD_IN_BREADSCRUMB',
            path: 'space/1/action/4',
            label: 'Action 4',
            changes: {action: 4}
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
        expect(store.state.global.breadscrumbs.length).toBe(3);
        onClickBreadScrumb(router, store, {position: 1, changes: {action: 2}, path: '/space/1/action/2'})
        expect(store.state.global.breadscrumbs.length).toBe(1);
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render change view without menuId', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space)
        });
        store.commit({type: 'REPLACE_CHANGE', changes: {test: 1}});
        router.push({path: '/'});
        expect(store.state.data.changes.test).toBe(1);
        expect(vm.$route.path).toBe('/');
        changeView(router, store, '1', null, '3', '4')
        expect(store.state.data.changes.test).toBe(undefined);
        expect(vm.$route.path).toBe('/space/1/action/3/view/4');
    });
    it('Render change view with menuId', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(Space)
        });
        store.commit({type: 'REPLACE_CHANGE', changes: {test: 1}});
        router.push({path: '/'});
        expect(store.state.data.changes.test).toBe(1);
        expect(vm.$route.path).toBe('/');
        changeView(router, store, '1', '2', '3', '4')
        expect(store.state.data.changes.test).toBe(undefined);
        expect(vm.$route.path).toBe('/space/1/menu/2/action/3/view/4');
    });
});
