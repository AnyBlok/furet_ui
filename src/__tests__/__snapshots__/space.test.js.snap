// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space component Render 1`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left">
                <!---->
                <nav class="breadcrumb">
                    <ul>
                        <li class="is-active"><a></a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <!---->
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`Space component Render breadscrumb 1`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left">
                <!---->
                <nav class="breadcrumb">
                    <ul>
                        <li class="is-active"><a>Action : 1</a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <div class="field has-addons">
                    <p class="control"><span data-label="List" class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><i class="fa fa-list"></i></span></a>
                        </span>
                    </p>
                </div>
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`Space component Render breadscrumb 2`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left">
                <!---->
                <nav class="breadcrumb">
                    <ul>
                        <li><a>
                                        Action 2
                                    </a></li>
                        <li class="is-active"><a>Action : 1</a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <div class="field has-addons">
                    <p class="control"><span data-label="List" class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><i class="fa fa-list"></i></span></a>
                        </span>
                    </p>
                </div>
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`Space component Render breadscrumb 3`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left">
                <!---->
                <nav class="breadcrumb">
                    <ul>
                        <li><a>
                                        Action 2
                                    </a></li>
                        <li><a>
                                        Action 3
                                    </a></li>
                        <li><a>
                                        Action 4
                                    </a></li>
                        <li class="is-active"><a>Action : 1</a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <div class="field has-addons">
                    <p class="control"><span data-label="List" class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><i class="fa fa-list"></i></span></a>
                        </span>
                    </p>
                </div>
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`Space component Render breadscrumb 4`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left">
                <!---->
                <nav class="breadcrumb">
                    <ul>
                        <li><a>
                                        Action 2
                                    </a></li>
                        <li class="is-active"><a>Action : 1</a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <div class="field has-addons">
                    <p class="control"><span data-label="List" class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><i class="fa fa-list"></i></span></a>
                        </span>
                    </p>
                </div>
                <!---->
            </div>
        </nav>
        <div class="columns is-gapless" style="margin-top:5px;">
            <!---->
            <div class="column" style="padding-left:10px;padding-right:10px;">
                <nav class="nav" style="background-color:inherit;">
                    <div class="nav-left">
                        <!---->
                        <nav class="breadcrumb">
                            <ul>
                                <li><a>
                                        Action 2
                                    </a></li>
                                <li class="is-active"><a></a></li>
                            </ul>
                        </nav>
                    </div>
                    <div class="nav-right">
                        <!---->
                        <!---->
                    </div>
                </nav>
                <!---->
            </div>
            <!---->
        </div>
    </div>
    <!---->
</div>
`;

exports[`Space component Render display left menu (close) 1`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left"><a class="button"><i aria-hidden="true" class="fa fa-bars fa-2x"></i></a>
                <nav class="breadcrumb">
                    <ul>
                        <li class="is-active"><a>Action : 1</a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <div class="field has-addons">
                    <p class="control"><span data-label="List" class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><i class="fa fa-list"></i></span></a>
                        </span>
                    </p>
                    <p class="control"><span data-label="Thumbnail" class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><i class="fa fa-th"></i></span></a>
                        </span>
                    </p>
                    <!---->
                </div>
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`Space component Render display left menu (open) 1`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <div class="column is-one-quarter is-half-mobile">
        <aside class="menu" style="padding:5px;">
            <ul class="menu-list">
                <li>
                    <!---->
                    <!----><a class="is-active"><span class="icon"><i class="fa fa-user"></i></span>
                    Customer
                </a></li>
                <li><span><span class="icon"><i class="fa fa-user"></i></span> Setting
                    </span>
                    <ul class="menu-list">
                        <li>
                            <!---->
                            <!---->
                            <a>
                                <!---->
                                Category
                            </a>
                        </li>
                        <li>
                            <!---->
                            <!---->
                            <a>
                                <!---->
                                Address
                            </a>
                        </li>
                    </ul>
                    <!---->
                </li>
            </ul>
        </aside>
    </div>
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left"><a class="button"><i aria-hidden="true" class="fa fa-bars fa-2x"></i></a>
                <nav class="breadcrumb">
                    <ul>
                        <li class="is-active"><a>Action : 1</a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <div class="field has-addons">
                    <p class="control"><span data-label="List" class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><i class="fa fa-list"></i></span></a>
                        </span>
                    </p>
                    <p class="control"><span data-label="Thumbnail" class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><i class="fa fa-th"></i></span></a>
                        </span>
                    </p>
                    <!---->
                </div>
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`Space component Render display right menu (close) 1`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left">
                <!---->
                <nav class="breadcrumb">
                    <ul>
                        <li class="is-active"><a>Action : 1</a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <div class="field has-addons">
                    <p class="control"><span data-label="List" class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><i class="fa fa-list"></i></span></a>
                        </span>
                    </p>
                    <p class="control"><span data-label="Thumbnail" class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><i class="fa fa-th"></i></span></a>
                        </span>
                    </p>
                    <!---->
                </div> <a class="button"><i aria-hidden="true" class="fa fa-bars fa-2x"></i></a></div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`Space component Render display right menu (open) 1`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left">
                <!---->
                <nav class="breadcrumb">
                    <ul>
                        <li class="is-active"><a>Action : 1</a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <div class="field has-addons">
                    <p class="control"><span data-label="List" class="tooltip is-info is-left is-medium"><a disabled="disabled" class="button is-primary"><span class="icon is-small"><i class="fa fa-list"></i></span></a>
                        </span>
                    </p>
                    <p class="control"><span data-label="Thumbnail" class="tooltip is-info is-left is-medium"><a class="button"><span class="icon is-small"><i class="fa fa-th"></i></span></a>
                        </span>
                    </p>
                    <!---->
                </div> <a class="button"><i aria-hidden="true" class="fa fa-bars fa-2x"></i></a></div>
        </nav>
        <!---->
    </div>
    <div class="column is-one-quarter is-half-mobile">
        <aside class="menu" style="padding:5px;">
            <ul class="menu-list">
                <li>
                    <!---->
                    <!----><a class="is-active"><span class="icon"><i class="fa fa-user"></i></span>
                    Customer
                </a></li>
                <li><span><span class="icon"><i class="fa fa-user"></i></span> Setting
                    </span>
                    <ul class="menu-list">
                        <li>
                            <!---->
                            <!---->
                            <a>
                                <!---->
                                Category
                            </a>
                        </li>
                        <li>
                            <!---->
                            <!---->
                            <a>
                                <!---->
                                Address
                            </a>
                        </li>
                    </ul>
                    <!---->
                </li>
            </ul>
        </aside>
    </div>
</div>
`;

exports[`SpaceMenu component Render 1`] = `<ul data-server-rendered="true" class="menu-list"></ul>`;

exports[`SpaceMenu component Render with submenu 1`] = `
<ul data-server-rendered="true" class="menu-list">
    <li>
        <!---->
        <!----><a class="is-active"><span class="icon"><i class="fa fa-user"></i></span>
                    Customer
                </a></li>
    <li><span><span class="icon"><i class="fa fa-user"></i></span> Setting
        </span>
        <ul class="menu-list">
            <li>
                <!---->
                <!---->
                <a>
                    <!---->
                    Category
                </a>
            </li>
            <li>
                <!---->
                <!---->
                <a>
                    <!---->
                    Address
                </a>
            </li>
        </ul>
        <!---->
    </li>
</ul>
`;

exports[`SpaceMenu component Render with submenu other selected menu 1`] = `
<ul data-server-rendered="true" class="menu-list">
    <li>
        <!---->
        <!----><a><span class="icon"><i class="fa fa-user"></i></span>
                    Customer
                </a></li>
    <li><span><span class="icon"><i class="fa fa-user"></i></span> Setting
        </span>
        <ul class="menu-list">
            <li>
                <!---->
                <!---->
                <a class="is-active">
                    <!---->
                    Category
                </a>
            </li>
            <li>
                <!---->
                <!---->
                <a>
                    <!---->
                    Address
                </a>
            </li>
        </ul>
        <!---->
    </li>
</ul>
`;

exports[`View Selector component Render 1`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left">
                <!---->
                <nav class="breadcrumb">
                    <ul>
                        <li class="is-active"><a></a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <!---->
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;

exports[`View Selector component Render unclickable selected 1`] = `
<div data-server-rendered="true" class="columns is-gapless" style="margin-top:5px;">
    <!---->
    <div class="column" style="padding-left:10px;padding-right:10px;">
        <nav class="nav" style="background-color:inherit;">
            <div class="nav-left">
                <!---->
                <nav class="breadcrumb">
                    <ul>
                        <li class="is-active"><a></a></li>
                    </ul>
                </nav>
            </div>
            <div class="nav-right">
                <!---->
                <!---->
            </div>
        </nav>
        <!---->
    </div>
    <!---->
</div>
`;
